Geometry/GEMGeometryBuilder/src/ME0GeometryBuilder.cc

```diff
@@ -0,0 +1,544 @@
+  LogTrace("ME0GeometryBuilder") << " dpar is vector with size = " << dpar.size() << std::endl;
+  for (unsigned int i = 0; i < dpar.size(); ++i) {
+    LogTrace("ME0GeometryBuilder") << " dpar [" << i << "] = " << dpar[i] / 10 << " cm " << std::endl;
+  }
+  LogTrace("ME0GeometryBuilder") << "size  b: " << b << "cm, B: " << B << "cm,  L: " << L << "cm, t: " << t << "cm "
+                                 << std::endl;
+#endif
+
+  bool isOdd = false;  // detId.chamber()%2;
+  ME0BoundPlane surf(boundPlane(fv, new TrapezoidalPlaneBounds(b, B, L, t), isOdd));
+  ME0Layer* layer = new ME0Layer(detId.layerId(), surf);
+  return layer;
+}
+
+ME0EtaPartition* ME0GeometryBuilder::buildEtaPartition(DDFilteredView& fv, ME0DetId detId) const {
+  LogTrace("ME0GeometryBuilder") << "buildEtaPartition " << fv.logicalPart().name().name() << " " << detId << std::endl;
+
+  // EtaPartition specific parameter (nstrips and npads)
+  DDValue numbOfStrips("nStrips");
+  DDValue numbOfPads("nPads");
+  std::vector<const DDsvalues_type*> specs(fv.specifics());
+  std::vector<const DDsvalues_type*>::iterator is = specs.begin();
+  double nStrips = 0., nPads = 0.;
+  for (; is != specs.end(); is++) {
+    if (DDfetch(*is, numbOfStrips))
+      nStrips = numbOfStrips.doubles()[0];
+    if (DDfetch(*is, numbOfPads))
+      nPads = numbOfPads.doubles()[0];
+  }
+  LogTrace("ME0GeometryBuilder") << ((nStrips == 0.) ? ("No nStrips found!!")
+                                                     : ("Number of strips: " + std::to_string(nStrips)));
+  LogTrace("ME0GeometryBuilder") << ((nPads == 0.) ? ("No nPads found!!")
+                                                   : ("Number of pads: " + std::to_string(nPads)));
+
+  // EtaPartition specific parameter (size)
+  std::vector<double> dpar = fv.logicalPart().solid().parameters();
+  double b = dpar[4] / cm;  // half bottom edge
+  double B = dpar[8] / cm;  // half top edge
+  double L = dpar[0] / cm;  // half apothem
+  double t = dpar[3] / cm;  // half thickness
+
+#ifdef EDM_ML_DEBUG
+  LogTrace("ME0GeometryBuilder") << " name of logical part = " << fv.logicalPart().name().name() << std::endl;
+  LogTrace("ME0GeometryBuilder") << " dpar is vector with size = " << dpar.size() << std::endl;
+  for (unsigned int i = 0; i < dpar.size(); ++i) {
+    LogTrace("ME0GeometryBuilder") << " dpar [" << i << "] = " << dpar[i] / 10 << " cm " << std::endl;
+  }
+  LogTrace("ME0GeometryBuilder") << "size  b: " << b << "cm, B: " << B << "cm,  L: " << L << "cm, t: " << t << "cm "
+                                 << std::endl;
+#endif
+
+  std::vector<float> pars;
+  pars.emplace_back(b);
+  pars.emplace_back(B);
+  pars.emplace_back(L);
+  pars.emplace_back(nStrips);
+  pars.emplace_back(nPads);
+
+  bool isOdd =
+      false;  // detId.chamber()%2; // this gives the opportunity (in future) to change the face of the chamber (electronics facing IP or electronics away from IP)
```

Comment:
put comment on separate line to avoid awkward line break from formatting

Reply:
@kpedro88 : done